internal/csi/csi.go

```diff
@@ -71,3 +82,68 @@ func BinsForRange(start, end uint32, minShift, depth int32) []uint16 {
 func maximumBinWidth(minShift, depth int32) uint32 {
 	return uint32(1 << uint32(minShift+depth*3))
 }
+
+// Read reads index data from CSI and returns a set of BGZF chunks covering
+// the header and all mapped reads that fall inside the specified region.  The
+// first chunk is always the BCF header.
+func Read(r io.Reader, region genomics.Region) ([]*bgzf.Chunk, error) {
+	csi, err := gzip.NewReader(r)
+	if err != nil {
+		return nil, fmt.Errorf("initializing gzip reader: %v", err)
+	}
+	defer csi.Close()
+	if err := binary.ExpectBytes(csi, []byte(csiMagic)); err != nil {
+		return nil, fmt.Errorf("checking magic: %v", err)
+	}
+
+	var csiHeader struct {
+		MinimumWidth   int32
+		Depth          int32
+		AuxilaryLength int32
+	}
+	if err := binary.Read(csi, &csiHeader); err != nil {
+		return nil, fmt.Errorf("reading the csi header: %v", err)
+	}
+	if _, err := io.CopyN(ioutil.Discard, csi, int64(csiHeader.AuxilaryLength)); err != nil {
+		return nil, fmt.Errorf("reading past auxiliary data: %v", err)
+	}
+
+	bcfHeader := &bgzf.Chunk{End: bgzf.LastAddress}
+	chunks := []*bgzf.Chunk{bcfHeader}
+	var refCount int32
+	if err := binary.Read(csi, &refCount); err != nil {
+		return nil, fmt.Errorf("reading the number of reference sequences: %v", err)
+	}
+	bins := BinsForRange(region.Start, region.End, csiHeader.MinimumWidth, csiHeader.Depth)
```

Comment:
Ideally we'd factor out the common code between here and the BAM reader.  The difficulty is the Offset field.  It's OK to submit this now, but maybe file an issue about it.  We could have some common function that takes a flag, for example, that would cover both.

Reply:
I've done it now so I don't commit duplicate code to begin with.